wdl-parsing/WDLParsingFunction/src/main/java/io/dockstore/wdlparser/App.java

```diff
@@ -0,0 +1,177 @@
+    Git.cloneRepository()
+        .setCloneAllBranches(false)
+        .setBranch(branch)
+        .setURI(uri)
+        .setDirectory(tempDirWithPrefix.toFile())
+        .call();
+    Path descriptorAbsolutePath = tempDirWithPrefix.resolve(descriptorRelativePathInGit);
+    String descriptorAbsolutePathString = descriptorAbsolutePath.toString();
+    LanguageParsingResponse response = getResponse(descriptorAbsolutePathString);
+    response.setLanguageParsingRequest(languageParsingRequest);
+    if (response.getSecondaryFilePaths() != null) {
+      response
+          .getSecondaryFilePaths()
+          .replaceAll(s -> s.replaceFirst(tempDirWithPrefix.toString(), ""));
+    }
+    return mapper.writeValueAsString(response);
+  }
+
+  // The first two lines aren't actual paths.
+  // It looks like "Success!" and "List of Workflow dependencies is:"
+  private static void handleSuccessResponse(
+      LanguageParsingResponse response, List<String> strings) {
+    strings.remove(0);
+    strings.remove(0);
+    // If there are no imports, womtool says None
+    if (strings.get(0).equals("None")) {
+      strings.remove(0);
+    }
+    response.setSecondaryFilePaths(strings);
+  }
+
+  /**
+   * Get a language parsing response by running womtool.
+   *
+   * @param descriptorAbsolutePathString Absolute path to the main descriptor file
+   * @return LanguageParsingResponse constructed after running womtool
+   */
+  public static LanguageParsingResponse getResponse(String descriptorAbsolutePathString) {
+    LanguageParsingResponse response = new LanguageParsingResponse();
+    response.setClonedRepositoryAbsolutePath(descriptorAbsolutePathString);
+    List<String> commandLineArgs = Arrays.asList("validate", "-l", descriptorAbsolutePathString);
+    try {
+      WomtoolMain.Termination termination =
+          WomtoolMain.runWomtool(
+              JavaConverters.collectionAsScalaIterableConverter(commandLineArgs).asScala().toSeq());
+      Option<String> stdout = termination.stdout();
+      // The womtool successful response returns lines of text.
+      String[] split = stdout.get().split(System.lineSeparator());
+      List<String> strings = new ArrayList<>();
+      Collections.addAll(strings, split);
+
+      // The first two lines aren't actual paths.
+      VersionTypeValidation versionTypeValidation = new VersionTypeValidation();
+      if (strings.get(0).equals("Success!")
+          && strings.get(1).equals("List of Workflow dependencies is:")) {
+        versionTypeValidation.setValid(true);
+        response.setVersionTypeValidation(versionTypeValidation);
+        handleSuccessResponse(response, strings);
+      } else {
+        versionTypeValidation.setValid(false);
+        response.setVersionTypeValidation(versionTypeValidation);
+      }
+      return response;
+    } catch (StackOverflowError e) {
+      VersionTypeValidation versionTypeValidation = new VersionTypeValidation();
+      versionTypeValidation.setValid(false);
```

Comment:
Shouldn't we be capturing the error via `VersionTypeValidation.setMessage`?

Reply:
wasn't added in this PR because it's non-trival